### PR TITLE
[sdl2] update to 2.0.22

### LINKED ISSUE #24654: [sdl2] update to 2.0.22
Library name: sdl2

New version number: 2.0.22, released April 25th 2022

Other information that may be useful (release notes, etc...)
Release notes: https://github.com/libsdl-org/SDL/releases/tag/release-2.0.22

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #24654 

Update sdl2 to the latest version 2.0.22

All features are tested successfully in the following triplet:
- x64-windows
- x64-windows-static

### CODE REVIEW

File: ports/sdl2/portfile.cmake [hunk 1]
```diff
@@ -2,17 +2,14 @@ set(SDL2_VERSION 2.0.20)
...
+    SHA512 809ac18aeb739cfc6558dc11a7b3abbdad62a8e651ae2bfc157f26947b0df063a34c117ea8bd003428b5618fa6ce21a655fda073f1ef15aa34bc4a442a578523
     HEAD_REF master
     PATCHES
         0001-sdl2-Enable-creation-of-pkg-cfg-file-on-windows.patch
         0002-sdl2-skip-ibus-on-linux.patch
         0003-sdl2-disable-sdlmain-target-search-on-uwp.patch
         0004-Define-crt-macros.patch
-        0005-Fix-uwp-joystick.patch
-        0006-Update-SDL_sysurl.cpp.patch
-        0007-timer-Fix-Emscripten-declaration-after-statement-err.patch
```
Comment (JonLiu1993) [inline ports/sdl2/portfile.cmake:15]: These three patches have been merged into the latest version

**Review by FrankXie05 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**